src/main/java/eu/ha3/presencefootsteps/world/GolemLookup.java

```diff
@@ -1,19 +1,20 @@
 package eu.ha3.presencefootsteps.world;
 
-import java.util.LinkedHashMap;
-import java.util.Map;
-import java.util.Set;
-
+import it.unimi.dsi.fastutil.objects.Object2ObjectLinkedOpenHashMap;
 import net.minecraft.entity.EntityType;
 import net.minecraft.util.Identifier;
 
-public class GolemLookup implements Lookup<EntityType<?>> {
+import java.util.Map;
+import java.util.Set;
 
-    private final Map<String, Map<Identifier, String>> substrates = new LinkedHashMap<>();
+public record GolemLookup(Map<String, Map<Identifier, String>> substrates) implements Lookup<EntityType<?>> {
```

Comment:
Not everything has to be a record

Reply:
Records are usually faster than normal classes and they should be used when possible

Reply:
Records are normal classes with an implicit all-args-constructor and overrides for equals(), hashCode(), and toString(), none of which are particularly required for this class. This class has one field and it's assigned immediately at construction, which I generally do not use records for.

Reply:
I understand, I'm going to revert this👍